application/src/main/java/org/togetherjava/tjbot/features/help/HelpSystemHelper.java

```diff
@@ -131,6 +138,71 @@ private RestAction<Message> sendExplanationMessage(GuildMessageChannel threadCha
         return action.setEmbeds(embeds);
     }
 
+    /**
+     * Gather first message from thread and pass into asynchronous sendChatGptAttempt.
+     * 
+     * @param threadChannel - The forum thread channel where the question originated
+     * @return A response from the AI service indicating success or failure.
+     * @see HelpSystemHelper#sendChatGptAttempt
+     */
+    RestAction<Message> prepareChatGptAttempt(ThreadChannel threadChannel) {
+        return threadChannel.retrieveMessageById(threadChannel.getIdLong())
+            .flatMap(message -> sendChatGptAttempt(message.getContentRaw(), threadChannel));
+    }
+
+    /**
+     * Determine between the title of the thread and the first message which to send to the AI. It
+     * uses a simple heuristic of length to determine if enough context exists in a question. If the
+     * title is used, it must also include a question mark since the title is often used more as an
+     * indicator of topic versus a question.
+     * 
+     * @param questionFirstMessage - The first message of the thread which originates from the
+     *        question asker.
+     * @param threadChannel - The thread in which the question was asked.
+     * @return An answer for the user from the AI service or a message indicating either an error or
+     *         why the message wasn't used.
+     */
+    private RestAction<Message> sendChatGptAttempt(String questionFirstMessage,
+            ThreadChannel threadChannel) {
+        String questionTitle = threadChannel.getName();
+        StringBuilder stringBuilder = new StringBuilder(questionFirstMessage);
+        List<ForumTag> tags = threadChannel.getAppliedTags();
+
+        if (questionFirstMessage.length() > MAX_QUESTION_LENGTH
+                || questionFirstMessage.length() < MIN_QUESTION_LENGTH) {
+            if (questionTitle.length() < MIN_QUESTION_LENGTH || !questionTitle.contains("?")) {
+                return threadChannel.sendMessage(String.format(
+                        """
+                                Your first message or title did not fit the format to be sent to ChatGPT. \
+                                Your message needs to be between %s and %s characters. If your question is longer than %s \
+                                then your title needs to include a question mark."
+                                    """,
+                        MIN_QUESTION_LENGTH, MAX_QUESTION_LENGTH, MAX_QUESTION_LENGTH));
+            }
+            stringBuilder.replace(0, stringBuilder.length(), questionTitle);
+        }
+
+        for (ForumTag tag : tags) {
+            stringBuilder.insert(0, String.format("%s ", tag.getName()));
+        }
+
+        String question = stringBuilder.toString();
+        String response = """
```

Comment:
might be helpful to add sth like `To continue talking to the AI, you can use /chatgpt.` (see our `mentionSlashCommand(...)` helper method)

Reply:
I am having some trouble with this change given that mentionSlashCommand requires a message. My main issue is twofold: designing the response (should the AI explanation message come first or second? Should it be an embed?). My other issue is just working with JDA and the RestMessage class in general and getting messages sent. Is there any documentation on how to do this? If how I have just changed it isn't passable (and I suspect it isn't), I will have to ask in questions (I can taste the irony) on how to grok this out.

Reply:
The Tj-Bot explanation can be a message, the chat-gpt response an attached embed, i guess. But you are free to design it as you like, if it looks bad otherwise 🤷

Yeah, we can discuss it in #questions :)

Reply:
The way I had it in mind originally, was like:

"Here is an AI assisted attempt blablabla use /chatgpt to follow up blablabla:"
> I am the chatgpt answer, in an embed, hello